dev/integration_tests/release_smoke_test/lib/main.dart

```diff
@@ -2,18 +2,29 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:convert' show utf8;
+import 'dart:typed_data';
+
+import 'package:flutter/foundation.dart';
 import 'package:flutter/widgets.dart';
 
-void main() {
+Future<void> main() async {
   const Text text = Text('Hello, world!', textDirection: TextDirection.ltr);
   // These calls must not result in an error. They behave differently in
   // release mode compared to debug or profile.
   // The test will grep logcat for any errors emitted by Flutter.
   print(text.toDiagnosticsNode());
   print(text.toStringDeep());
+  // regression test for https://github.com/flutter/flutter/issues/49601
+  final List<int> computed = await compute(_utf8encode, 'test', debugLabel: null);
```

Comment:
Is `_utf8encode` necessary? `await compute(utf8.encode, 'test', debugLabel: null)`

Reply:
Done

Reply:
I guess it is necessary.  Test fails without it this way.

Reply:
>I guess it is necessary. Test fails without it this way.

Quite surprising

Reply:
Outch. It's really unfortunate. It looks like a bug to me.

```
E/flutter ( 9950): [ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: Invalid argument(s): Illegal argument in isolate message : (object is a closure - Function 'encode':.)
```

Reply:
I think it has something to do with the way tear offs are handled by the VM ... @chinmaygarde or @mraleph would probably know

Reply:
Is there a reduced test case for the failure?

Reply:
Call `compute` with utf8.encode as the first argument

Reply:
This is a limitation of Dart isolate ports (albeit not a well documented one) - you can only send top-level tear-offs, not instance tear-offs or arbitrary closures through ports.

That is why `_utf8encode` works (it is a top level tear off) and `utf8.encode` fails. 

I have filed https://github.com/dart-lang/sdk/issues/40370 to make executive decision on finally lifting this restriction.